runtime/src/snapshot_minimizer.rs

```diff
@@ -0,0 +1,609 @@
+use {
+    crate::{
+        accounts_db::{AccountStorageEntry, AccountsDb, PurgeStats},
+        bank::Bank,
+    },
+    dashmap::DashSet,
+    log::info,
+    rayon::{
+        iter::{IntoParallelIterator, IntoParallelRefIterator, ParallelIterator},
+        prelude::ParallelSlice,
+    },
+    solana_measure::measure,
+    solana_sdk::{
+        account::ReadableAccount,
+        account_utils::StateMut,
+        bpf_loader_upgradeable::{self, UpgradeableLoaderState},
+        clock::Slot,
+        pubkey::Pubkey,
+    },
+    std::{
+        collections::HashSet,
+        sync::{
+            atomic::{AtomicUsize, Ordering},
+            Arc, Mutex,
+        },
+    },
+};
+
+pub struct SnapshotMinimizer {
```

Comment:
/// brief desc - like just local to this file to make it easy to modify the bank with a minimal set of accounts